File: src/main/java/komu/demodel/domain/ClassDirectoryInputSource.java
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

import komu.demodel.utils.Check;
import komu.demodel.utils.ExtensionFileFilter;
import komu.demodel.utils.FileResource;
import komu.demodel.utils.FileSet;
import komu.demodel.utils.Resource;
import komu.demodel.utils.ResourceProvider;

public final class ClassDirectoryInputSource implements InputSource {

    private final File directory;

    public ClassDirectoryInputSource(File directory) {
        Check.notNull(directory, "directory");

        this.directory = directory;
    }

    @Override
    public ResourceProvider getResources() {
        final List<Resource> resources = new ArrayList<Resource>();

        for (File file : new FileSet(directory, new ExtensionFileFilter("class")))
            resources.add(new FileResource(file));

        return new ResourceProvider() {
            @Override
            public Iterator<Resource> iterator() {
                return resources.iterator();
            }
            
            @Override
            public void close() throws IOException {
            }
        };
    }
    
    @Override
    public String toString() {
        return directory.toString();
    }
}

File: src/main/scala/komu/demodel/domain/project/ClassDirectoryInputSource.scala
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain.project

import java.io.File
import komu.demodel.utils.{ Resource, FileSet, ExtensionFileFilter, FileResource }

final class ClassDirectoryInputSource(directory: File) extends InputSource {
  assert(directory != null)

  def withResources(thunk: Resource => Unit) {
    for (file <- new FileSet(directory, new ExtensionFileFilter("class")))
      thunk(new FileResource(file))
  }
    
  override def toString = directory.toString
}
